src/htm/ntypes/Value.hpp

```diff
@@ -14,122 +16,236 @@
  * You should have received a copy of the GNU Affero Public License
  * along with this program.  If not, see http://www.gnu.org/licenses.
  * --------------------------------------------------------------------- */
-
-/** @file
- * Definitions for the Value class
- *
- * A Value object holds a Scalar or an Array
- * A ValueMap is essentially a map<string, Value>
- * It is used internally in the conversion of YAML strings to C++ objects.
- * The API and implementation are geared towards clarify rather than
- * performance, since it is expected to be used only during network construction.
- */
-
 #ifndef NTA_VALUE_HPP
 #define NTA_VALUE_HPP
 
-
-#include <map>
+#include <htm/utils/Log.hpp>
 #include <string>
+#include <vector>
+#include <map>
+#include <memory>
+#include <iostream>
+#include <sstream>
+#include <algorithm>  // transform
+#include <cerrno>
+#include <cstring>    // std::strerror(errno)
```

Comment:
nit, this seems to define a lot of headers, is all of that really needed, or can some be moved to cpp?

Reply:
I am sure I can remove some when I clean up.